### PR TITLE
New Package: bml

### PR DESCRIPTION
**Pull request by rspavel**
Added LANL's BML library as Spack Package as dependency for future
packages

### CODE REVIEW

File: var/spack/repos/builtin/packages/lanl-bml/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+class LanlBml(CMakePackage):
+    """The basic matrix library (bml) is a collection of various matrix data
+    formats (in dense and sparse) and their associated algorithms for basic
+    matrix operations."""
+
+    homepage = "https://github.com/qmmd/bml"
+    url      = "https://github.com/qmmd/bml"
+
+    version('master', git='https://github.com/qmmd/bml', branch='master')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/lanl-bml/package.py]: Can you add at least one version that points to a release?  And call this version `develop`?  Spack prefers to check out the latest non-develop version, and we like that to be something fixed and verifiable.

**Comment by adamjstewart** [comment]
@tgamblin Can you approve?

**Comment by adamjstewart** [comment]
One last question: should the package be called `lanl-bml` or just `bml`? Does anyone refer to it as `lanl-bml` or is LANL just the developer? Are there any other `bml` packages out there?

**Comment by rspavel** [comment]
@adamjstewart Sorry. The package is bml and to my knowledge it has always been referred to as such, but is also a very generic name. A quick google suggests there are matlab packages as well as  an ubuntu package.

I have no issue renaming the spackage to bml and taking advantage of us being first (possibly only) and can make that change if preferred.

**Comment by adamjstewart** [comment]
@rspavel Let's do it. We can rename it later if necessary.

**Comment by rspavel** [comment]
@adamjstewart Done

**Pull request merged**